cdbot/cogs/fun.py

```diff
@@ -324,19 +324,23 @@ def check(reaction, user):
         """
         Adds a quote message ID to the database, and attempts to identify the author of the quote.
         """
+        author_id = None
         if quote.author.id == QUOTES_BOT_ID:
             if not quote.embeds:
                 return
             embed = quote.embeds[0]
-            author_id = int(embed.author.icon_url.split("/")[4])
-            try:
-                await self.bot.fetch_user(author_id)
-            except NotFound:
-                author_info = embed.author.split("#")
-                author_id = discord_find(
-                    lambda m: m.name == author_info[0] or m.discriminator == author_info[1],
+            icon_url = embed.author.icon_url
+            if type(icon_url) == embeds._EmptyEmbed or 'twimg' in icon_url:
```

Comment:
```suggestion
            if type(icon_url) == Embed.Empty or 'twimg' in icon_url:
```

Reply:
It should be able to be referred to inside the Empty class.

Reply:
`py
@bot.command()
async def t(ctx, mid: int):
    channel = bot.get_channel(463657120441696256)
    msg = await channel.history(limit=None).get(id=mid)
    icon_url = msg.embeds[0].author.icon_url
    if type(icon_url) == discord.Embed.Empty:
        print("discord.Embed.Empty")
    if type(icon_url) == discord.embeds._EmptyEmbed:
        print("discord.embeds._EmptyEmbed")`
Running this code on one of the problematic quotes only prints `discord.embeds._EmptyEmbed`, it will not work with `Embed.Empty`.

Reply:
Fair enough.